### PR TITLE
refactor(core): create generic sort operations

### PR DESCRIPTION
**Pull request by CaerusKaru**

### CODE REVIEW

File: src/lib/core/media-marshaller/media-marshaller.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {merge, Observable, Subject, Subscription} from 'rxjs';
 import {filter, tap} from 'rxjs/operators';
 
 import {BreakPoint} from '../breakpoints/break-point';
-import {sortDescendingPriority} from '../breakpoints/breakpoint-tools';
+import {sortDescendingPriority} from '../utils/sort';
```
[the remark this answers is not in the document]
Comment (CaerusKaru) [reply src/lib/core/media-marshaller/media-marshaller.ts]: Nope!

File: src/lib/core/utils/sort.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+interface WithPriority {
+  priority?: number;
+}
+
+/** HOF to sort the breakpoints by descending priority */
+export function sortDescendingPriority<T extends WithPriority>(a: T | null, b: T | null): number {
```
[the remark this answers is not in the document]
Comment (CaerusKaru) [reply src/lib/core/utils/sort.ts]: Bc if it’s just `WithPriority` then the interface becomes too restrictive. Or did you have something else in mind?

File: src/lib/core/media-observer/media-observer.ts [hunk 3]
```diff
@@ -78,6 +74,10 @@ export class MediaObserver {
               protected matchMedia: MatchMedia,
               protected hook: PrintHook) {
     this._media$ = this.watchActivations();
+    this.media$ = this._media$.pipe(
```
[the remark this answers is not in the document]
Comment (CaerusKaru) [reply src/lib/core/media-observer/media-observer.ts]: The interface for `asObservable` is unaffected by this change.

File: src/lib/core/utils/sort.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThomasBurleson) [reply src/lib/core/utils/sort.ts]: I was not aware of this trick. Nice.

**Pull request merged**